bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/CallOptionsFactory.java

```diff
@@ -71,17 +77,28 @@ public ConfiguredCallOptionsFactory(CallOptionsConfig config) {
     }
 
     @Override
-    public <RequestT> CallOptions create(
-        MethodDescriptor<RequestT, ?> descriptor, RequestT request) {
-      if (!config.isUseTimeout() || request == null) {
+    /**
+     * Creates a {@link CallOptions} with a focus on {@link Deadlines}.  Deadlines are decided in the following order:
+     * <ol>
+     *     <li> If a user set a  {@link Context} deadline (see {@link Context#getDeadline()}), use that</li>
+     *     <li> If a user configured deadlines via {@link CallOptionsConfig}, use it.</li>
+     *     <li> Otherwise, use {@link CallOptions#DEFAULT}.</li>
+     * </ol>
+     *
+     */
+    public <RequestT> CallOptions create(MethodDescriptor<RequestT, ?> descriptor, RequestT request) {
+      Deadline contextDeadline = Context.current().getDeadline();
```

Comment:
How does the deadline in Context set by user?
It seems to be harder to track where the deadline is coming from.

Reply:
See the test for how a user would set the context deadline.  Where the operation deadline comes from is kind of irrelevant.